app/javascript/components/visualization/DotPlot.js

```diff
@@ -60,20 +68,25 @@ export default function DotPlot({
 
   return (
     <div>
+      { ErrorComponent }
       { cluster &&
       <>
         <div id={graphId} className="dotplot-graph"></div>
-        <DotPlotLegend/>
+        { !showError && <DotPlotLegend/> }
       </> }
       { !cluster && <FontAwesomeIcon icon={faDna} className="gene-load-spinner"/> }
     </div>
   )
 }
 
+const SafeDotPlot = withErrorBoundary(DotPlot)
+export default SafeDotPlot
```

Comment:
Blocker: this naming pattern will confuse maintainers.  

The default export is `SafeDotPlot`, but it's imported as `DotPlot` -- the same name as the "unsafe" function in this module.  To avoid confusion over similar names, please align the imported and exported names -- e.g. name the unwrapped dot plot function "UnsafeDotPlot", then:

```suggestion
const DotPlot = withErrorBoundary(UnsafeDotPlot)
export default DotPlot
```

And similarly for the other visualization components.

This would also make it obvious when using the unsafe version of the function.